src/utils/validateInput.ts

```diff
@@ -0,0 +1,31 @@
+export type InputValidator<T> = (input: T) => boolean
+
+const passesAll = <T>(iv: InputValidator<T>[]) => (value: T) =>
+  iv.map(fn => fn(value)).reduce((p, c) => p && c, true)
```

Comment:
Can use [Array.every](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/every) instead of map + reduce.

Reply:
ah of course, I knew there was something obvious I was missing